packages/pdf/src/RPA/PDF/keywords/model.py

```diff
@@ -0,0 +1,585 @@
+        self.pageid = pageid
+        self.bbox = iterable_items_to_int(bbox)
+        self.rotate = rotate
+        self.content = OrderedDict()
+        self.content_id = 0
+
+    def add_content(self, content: Any) -> None:
+        self.content[self.content_id] = content
+        self.content_id += 1
+
+    def get_content(self) -> OrderedDict:
+        return self.content
+
+    def get_figures(self) -> OrderedDict:
+        return {k: v for k, v in self.content.items() if isinstance(v, RpaFigure)}
+
+    def get_textboxes(self) -> OrderedDict:
+        return {k: v for k, v in self.content.items() if isinstance(v, RpaTextBox)}
+
+    def __str__(self) -> str:
+        page_as_str = '<page id="%s" bbox="%s" rotate="%d">\n' % (
+            self.pageid,
+            bbox2str(self.bbox),
+            self.rotate,
+        )
+        for _, c in self.content.items():
+            page_as_str += f"{c}\n"
+        return page_as_str
+
+
+class RpaPdfDocument:
+    """Class for parsed PDF document"""
+
+    encoding: str = "utf-8"
+    pages: OrderedDict
+    xml_content: bytearray = bytearray()
+
+    def __init__(self) -> None:
+        self.pages = OrderedDict()
+
+    def append_xml(self, xml: bytes) -> None:
+        self.xml_content += xml
+
+    def add_page(self, page: RpaPdfPage) -> None:
+        self.pages[page.pageid] = page
+
+    def get_pages(self) -> OrderedDict:
+        return self.pages
+
+    def get_page(self, pagenum: int) -> RpaPdfPage:
+        return self.pages[pagenum]
+
+    def dump_xml(self) -> str:
+        return self.xml_content.decode("utf-8")
+
+
+class RPAConverter(PDFConverter):
+    """Class for converting PDF into RPA classes"""
+
+    CONTROL = re.compile("[\x00-\x08\x0b-\x0c\x0e-\x1f]")
+
+    def __init__(
+        self,
+        rsrcmgr,
+        codec="utf-8",
+        pageno=1,
+        laparams=None,
+        imagewriter=None,
+        stripcontrol=False,
+    ):
+        PDFConverter.__init__(
+            self, rsrcmgr, sys.stdout, codec=codec, pageno=pageno, laparams=laparams
+        )
+        self.rpa_pdf_document = RpaPdfDocument()
+        self.figure = None
+        self.current_page = None
+        self.imagewriter = imagewriter
+        self.stripcontrol = stripcontrol
+        self.write_header()
+
+    def write(self, text):
+        if self.codec:
+            text = text.encode(self.codec)
+        self.rpa_pdf_document.append_xml(text)
+
+    def write_header(self):
+        if self.codec:
+            self.write('<?xml version="1.0" encoding="%s" ?>\n' % self.codec)
+        else:
+            self.write('<?xml version="1.0" ?>\n')
+        self.write("<pages>\n")
+
+    def write_footer(self):
+        self.write("</pages>\n")
+
+    def write_text(self, text):
+        if self.stripcontrol:
+            text = self.CONTROL.sub("", text)
+        self.write(enc(text))
+
+    def receive_layout(self, ltpage):  # noqa: C901 pylint: disable=R0915
```

Comment:
This function sure is... something.

Reply:
This is some kind of pdfminer magic method... need documenting